rosbag2_transport/test/rosbag2_transport/test_record.cpp

```diff
@@ -151,9 +151,14 @@ TEST_F(RecordIntegrationTestFixture, receives_latched_messages)
   // Publish messages before starting recording
   test_msgs::msg::Strings msg;
   msg.string_value = "Hello";
-  for (size_t i = 0; i < num_latched_messages; i++) {
-    publisher_transient_local->publish(msg);
-    rclcpp::spin_some(publisher_node);
+  {
+    rclcpp::executors::SingleThreadedExecutor exec;
+    exec.add_node(publisher_node);
+    for (size_t i = 0; i < num_latched_messages; i++) {
+      publisher_transient_local->publish(msg);
+      exec.spin_some();
+    }
+    exec.remove_node(publisher_node);
```

Comment:
Oh, since you put the executor in a scope, this is probably redundant now.  But it doesn't hurt anything, so I'm fine if you want to keep it.